group-coordinator/src/main/java/org/apache/kafka/coordinator/group/Result.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * The result of an operation applied to a state machine. The result
+ * contains a list of {{@link Record}} and a response.
```

Comment:
So the idea is that an operation will apply records to the log and send a response about the result?

Reply:
This is correct. Every operation generates records and a response. The records are applied to the log and the response is returned when they are committed.